File: app/src/main/java/ru/merkulyevsasha/news/data/db/mappers/ArticleMapper.java
package ru.merkulyevsasha.news.data.db.mappers;

import ru.merkulyevsasha.news.data.Mapper;
import ru.merkulyevsasha.news.data.db.entities.ArticleEntity;
import ru.merkulyevsasha.news.pojos.Article;

public class ArticleMapper implements Mapper<Article, ArticleEntity> {
    @Override
    public ArticleEntity map(Article item) {
        ArticleEntity entity = new ArticleEntity();
        entity.setCategory(item.getCategory());
        entity.setDescription(item.getDescription());
        entity.setLink(item.getLink());
        entity.setPictureUrl(item.getPictureUrl());
        entity.setPubDate(item.getPubDate());
        entity.setSourceNavId(item.getSourceNavId());
        entity.setTitle(item.getTitle());
        entity.setSearch(item.getSearch());
        return entity;
    }
}

File: app/src/main/java/ru/merkulyevsasha/news/data/db/mappers/ArticleMapper.kt
package ru.merkulyevsasha.news.data.db.mappers

import ru.merkulyevsasha.news.data.Mapper
import ru.merkulyevsasha.news.data.db.entities.ArticleEntity
import ru.merkulyevsasha.news.models.Article

class ArticleMapper : Mapper<Article, ArticleEntity> {
    override fun map(item: Article): ArticleEntity {
        return ArticleEntity(
                item.sourceNavId,
                item.title,
                item.link,
                item.description ?: "",
                item.pubDate,
                item.category ?: "",
                item.search,
                item.pictureUrl ?: ""
        )
    }
}
